frontend/ui/widget/timewidget.lua

```diff
@@ -0,0 +1,354 @@
+local Blitbuffer = require("ffi/blitbuffer")
+local Button = require("ui/widget/button")
+local ButtonTable = require("ui/widget/buttontable")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local CloseButton = require("ui/widget/closebutton")
+local Device = require("device")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local Geom = require("ui/geometry")
+local GestureRange = require("ui/gesturerange")
+local Font = require("ui/font")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local HorizontalSpan = require("ui/widget/horizontalspan")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local LineWidget = require("ui/widget/linewidget")
+local OverlapGroup = require("ui/widget/overlapgroup")
+local TextBoxWidget = require("ui/widget/textboxwidget")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local VerticalSpan = require("ui/widget/verticalspan")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+local Screen = Device.screen
+
+local TimeWidget = InputContainer:new{
+    title_face = Font:getFace("x_smalltfont"),
+    width = nil,
+    height = nil,
+    curr_hour = 0,
```

Comment:
Why not a more neutral, simpler name like just `hour` and `min`? This widget offers great potential for anything that requires some form of time input. That could be "the time is currently set to 12:16", but also "currently set to display a notification at 12:30" or "currently set to display a notification at 1:15 minutes from now."